query/src/main/scala/filodb/query/exec/SetOperatorExec.scala

```diff
@@ -0,0 +1,142 @@
+package filodb.query.exec
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+import monix.reactive.Observable
+
+import filodb.core.metadata.Dataset
+import filodb.core.query._
+import filodb.memory.format.{ZeroCopyUTF8String => Utf8Str}
+import filodb.memory.format.ZeroCopyUTF8String._
+import filodb.query._
+import filodb.query.BinaryOperator.{LAND, LOR, LUnless}
+
+/**
+  * Set operator between results of lhs and rhs plan.
+  *
+  * This ExecPlan accepts two sets of RangeVectors lhs and rhs from child plans.
+  * It then does a join of the RangeVectors based on the fields of their keys as
+  * dictated by `on` or `ignoring` fields passed as params.
+  *
+  * Joins can be Many-to-Many.
+  *
+  * @param lhs ExecPlan that will return results of LHS expression
+  * @param rhs ExecPlan that will return results of RHS expression
+  * @param binaryOp the binary operator
+  * @param cardinality the cardinality of the join relationship as a hint
+  * @param on fields from range vector keys to include while performing the join
+  * @param ignoring fields from range vector keys to exclude while performing the join
+  */
+final case class SetOperatorExec(id: String,
+                                dispatcher: PlanDispatcher,
+                                lhs: Seq[ExecPlan],
+                                rhs: Seq[ExecPlan],
+                                binaryOp: BinaryOperator,
+                                cardinality: Cardinality,
+                                on: Seq[String],
+                                ignoring: Seq[String]) extends NonLeafExecPlan {
+  require(cardinality == Cardinality.ManyToMany, "set operations must only use many-to-many matching")
```

Comment:
IS `cardinality` even needed now ? Can be removed.

Reply:
Prometheus has this validation that set operations must only use many-to-many matching

Reply:
Removed cardinality. Validation is done in vectorMatch validate